### PR TITLE
[DLR] Create a new DlrPredictor to copy the model

### PR DESCRIPTION
**Pull request by stu1130**
## Description ##

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: dlr/dlr-engine/src/main/java/ai/djl/dlr/engine/DlrModel.java [hunk 1]
```diff
@@ -32,6 +33,7 @@
  */
 public class DlrModel extends BaseModel {
 
+    private Device device;
```
Comment (frankfliu) [inline dlr/dlr-engine/src/main/java/ai/djl/dlr/engine/DlrModel.java:36]: We don't need this here, just call manager.getDevice() when needed.

File: examples/src/main/java/ai/djl/examples/inference/benchmark/util/AbstractBenchmark.java [hunk 2]
```diff
@@ -288,10 +288,7 @@ public NDList processInput(TranslatorContext ctx, Object input) {
                         /** {@inheritDoc} */
                         @Override
                         public Object processOutput(TranslatorContext ctx, NDList list) {
-                            FloatBuffer fb = list.get(0).toByteBuffer().asFloatBuffer();
-                            float[] ret = new float[fb.remaining()];
-                            fb.get(ret);
-                            return ret;
+                            return list.get(0).toType(DataType.FLOAT32, false);
```
Comment (frankfliu) [inline examples/src/main/java/ai/djl/examples/inference/benchmark/util/AbstractBenchmark.java:291]: toType() is not supported by Ort and Dlr

File: dlr/dlr-engine/src/main/java/ai/djl/dlr/engine/DlrPredictor.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+import ai.djl.Device;
+import ai.djl.dlr.jni.JniUtils;
+import ai.djl.inference.Predictor;
+import ai.djl.translate.Translator;
+
+/**
+ * {@code DlrPredictor} is special implementation of {@link Predictor} for DLR.
+ *
+ * <p>The native Dlr doesn't support multi-threading feature, when creating a new DlrPredictor, we
```
Comment (zachgk) [inline dlr/dlr-engine/src/main/java/ai/djl/dlr/engine/DlrPredictor.java:23]: Can this be done in the DlrModel? So, the model would track whether it is the first created predictor in the .newPredictor(...), and if not give it a new handle.

Comment (stu1130) [reply dlr/dlr-engine/src/main/java/ai/djl/dlr/engine/DlrPredictor.java:23]: Yes we can, but we need to override the getBlock to do implicitly copy, which is not ideal for end user. Let's say if user call DlrModel.getBlock() to play around the object they don't realize we reload another new TVM model.

**Review by frankfliu — APPROVED** [review]

**Pull request merged**